src/main/java/org/mqttbee/mqtt/message/publish/MqttTopicAliasMapping.java

```diff
@@ -45,14 +46,14 @@ public MqttTopicAliasMapping(final int topicAliasMaximum) {
 
     public int set(@NotNull final MqttTopicImpl topic, @NotNull final TopicAliasUsage topicAliasUsage) {
         int topicAlias = MqttStatefulPublish.DEFAULT_NO_TOPIC_ALIAS;
-        if (topicAliasUsage != TopicAliasUsage.MUST_NOT) {
-            if (nextTopicAlias > topicAliasMaximum) {
-                if (topicAliasUsage == TopicAliasUsage.MAY_OVERWRITE) {
-                    topicAlias = random.nextInt(topicAliasMaximum) + 1;
-                    hashMap.values().remove(topicAlias);
-                    hashMap.put(topic.toString(), topicAlias);
-                }
-            } else {
+        if (!unusedTopicAliasAvailable()) {
```

Comment:
If we are changing this now, I think the following is even simpler:
```
final int topicAlias = DEFAULT_NO_TOPIC_ALIAS;
if ((topicAliasUsage == YES) || (topicAliasUsage == IF_AVAILABLE)) {
    if (nextTopicAlias <= topicAliasMaximum) {
        topicAlias = nextTopicAlias++;
        hashMap.put(topic.toString(), topicAlias);
    } else if (topicAliasUsage == YES) {
        topicAlias = random.nextInt(topicAliasMaximum) + 1;
        hashMap.values().remove(topicAlias);
        hashMap.put(topic.toString(), topicAlias);
    }
}
return topicAlias;
```
I think it is easier to understand if we first handle the normal case (nextTopicAlias <= topicAliasMaximum) and then the overwriting. With this I think we can close the old PR #25, as this is very concise.

Reply:
DONE